### DISCUSSION TITLE
Not working from multi action button

### DISCUSSION THREAD

**Discussion by CommanderB** [ybizeul/StreamDeckWS#14] [Q&A]
I have the following flow - this has 3 buttons. Button 1 or 2 can be on, button 3 (bottom of the switch) turns them both off. 

![SD KeyOff](https://user-images.githubusercontent.com/18680188/104856793-798bf280-58e2-11eb-91a2-26184c0eb8e2.png)

This works as expected when button 3 is WSProxy action on the SD. 

It does not work when the WSProxy action is a part of a StreamDeck multi action. 

I have worked around this by migrating the other actions done in the multi action to node-red. But this might not always be an option.

**Comment by ybizeul** [comment]
The issue with multi-actions, is they are not displayed as traditional keys, I don't remember exactly what message they send.

It would be useful to add a debug node right after the `[ws]` and see what you get when you run the multi action.

**Reply by CommanderB** [reply]
There is a `TypeError` before the debug node fires. Three total for willAppear/keyDown/keyUp

![SD TypeError](https://user-images.githubusercontent.com/18680188/104953271-13f83e80-5994-11eb-856a-a2340d7fa621.png)

I have my workaround, so not urgent to fix for me

**Reply by ybizeul** [reply]
The `TypeError` appears for a multi-action if you do not have the `id` field set, would that be the case ?

That aside, if you catch `KeyUp` only, you should be able to make it work

**Comment by CommanderB** [comment]
I'm not sure the issue is with multi-action buttons, I am getting all sorts of odd errors trying to connect to more buttons. New thread started.
